x-pack/plugins/data_enhanced/server/search/session/monitoring_task.ts

```diff
@@ -48,12 +54,17 @@ function searchSessionRunner(
             logger,
           },
           sessionConfig
-        );
+        )
+          .pipe(takeUntil(aborted$.pipe(filter((aborted) => aborted))))
+          .toPromise();
 
         return {
           state: {},
         };
       },
+      cancel: async () => {
+        aborted$.next(true);
```

Comment:
@elastic/kibana-alerting-services, My assumption is that it works the following way (but honestly, from testing locally I am not seeing the behavior I am expecting to see): 

* Task scheduling inveral is `10s`
* Task timeout is `5m`
* Let's assume the task always takes longer than `5m` so it always will be canceled:

10s -- the task is scheduled
~5m10s -- the task is canceled
~5m20s -- new task is scheduled
~10m20s -- the task is canceled
~5m30s -- new task is scheduled


^^ If this is expected, then I am not seeing this. I see multiple tasks being overlapped and `canceled` not being called for each task. So I assume there is either a bug or some different logic in play.

Anyway, could you please review the API usage? And let me know if we should chat/debug deeper about that scenario I tried to test locally

Reply:
Just run my exercises locally again and everything seems alright :) 
probably was my mistake before